tests/integration/test_where.py

```diff
@@ -45,6 +45,98 @@ def test_condition(cond):
     assert np.array_equal(np.where(anp, xnp, ynp), num.where(a, x, y))
 
 
+@pytest.mark.parametrize(
+    "shape_a",
+    ((1,), (3,), (1, 3), (3, 3), (2, 3, 3)),
+    ids=lambda shape_a: f"(shape_a={shape_a})",
+)
+def test_broadcast(shape_a):
+    a = mk_seq_array(num, shape_a)
+    anp = mk_seq_array(np, shape_a)
+    cond = a > 5
+    cond_np = anp > 5
+
+    shape_x = (3, 3)
+    x = mk_seq_array(num, shape_x)
+    xnp = mk_seq_array(np, shape_x)
+    shape_y = (1, 3)
+    y = mk_seq_array(num, shape_y) * 10
+    ynp = mk_seq_array(np, shape_y) * 10
+
+    assert np.array_equal(np.where(cond_np, xnp, ynp), num.where(cond, x, y))
+
+
+@pytest.mark.xfail
+def test_condition_none():
+    # In Numpy, pass and returns [1, 2]
+    # In cuNumeric, raises AttributeError:
+    # 'NoneType' object has no attribute '_maybe_convert'
+    x = 0
+    ynp = np.array([1, 2])
+    y = num.array(ynp)
+    assert np.array_equal(np.where(None, x, ynp), num.where(None, x, y))
```

Comment:
Here the `cond` array is `None`, and is picked up by NumPy's default undocumented handling of `None`s. I would personally be happy if we switched our behavior to rejecting `None`s in `@add_boilerplate` ("expected something convertible to cunumeric.ndarray; got `None`"), even if it differs from NumPy's behavior.

Reply:
Or `NotImplementedError` if you prefer ("Cannot accept `None` because cuNumeric does not currently support arrays of object type").